### PR TITLE
Replace AppVeyor on GitHub Actions

### PR DESCRIPTION
**Pull request by Aleh-Yanushkevich**
- Implemented workflow for building/testing and publish code coverage results to Codecov during PR creation.
- Implemented workflow for publish nuget package to _**NuGet**_/_**MyGet**_ (NuGet by default) after release creation or run  manually.
- Removed AppVeyor yaml file.

Also should be performed next steps:

1. Add NUGET_API_KEY and NUGET_SOURCE secrets.
2. Add MYGET_API_KEY and MYGET_SOURCE secrets.

### CODE REVIEW

**Comment by nvborisenko** [comment]
- For `release` I propose to skip testing stage at all
- `release` triggers only on tag
- `ci` triggers on every commit
- NUGET_SOURCE is not a secret
- I propose pushing package to MyGet on every commit into `develop`

**Comment by Aleh-Yanushkevich** [comment]
@nvborisenko 
- release workflow was split on release and release-myget. I kept trigger for release workflow as is, cuz github actions doesn't support easy way for triggering workflow after adding a tag to specific branch. I think `on.release.published` is the best trigger here. 
- NuGet source has been hardcoded into workflow.

Should I keep MyGet source as a secret or not?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+on: push
+
+env:
+  CONFIGURATION: Release
+
+jobs:
+
+  buid:
+    runs-on: windows-2022
```
Comment (nvborisenko) [inline .github/workflows/ci.yml:11]: or just `latest`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    timeout-minutes: 10
```
Comment (nvborisenko) [inline .github/workflows/ci.yml:12]: I propose to leave default one rather than limiting us

File: .github/workflows/release-myget.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  push:
+    branches:
+      - develop
+     
+env:
+  CONFIGURATION: Release
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (nvborisenko) [inline .github/workflows/release-myget.yml:13]: On `ubuntu` it's not able to build .net framework target, please check.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+name: ci
+
+on: push
+
+env:
+  CONFIGURATION: Release
+
+jobs:
+
+  buid:
```
Comment (nvborisenko) [inline .github/workflows/ci.yml:10]: mistype

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Aleh-Yanushkevich) [reply .github/workflows/ci.yml]: Changed to latest.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Aleh-Yanushkevich) [reply .github/workflows/ci.yml]: Fixed.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (Aleh-Yanushkevich) [reply .github/workflows/ci.yml]: Fixed.

File: .github/workflows/release-myget.yml (hunk 3 above)
Comment (Aleh-Yanushkevich) [reply .github/workflows/release-myget.yml:13]: As I understand correctly, from .NET5 it should work (I've tested in on my prive project). dotnet/sdk#4009

**Review by nvborisenko — APPROVED** [review]

**Pull request merged**